libraries/AC_WPNav/AC_WPNav.cpp

```diff
@@ -1127,7 +839,50 @@ void AC_WPNav::wp_speed_update(float dt)
 
     // update position controller speed
     _pos_control.set_max_speed_xy(curr_max_speed_xy_cms);
-    
-    // flag that wp leash must be recalculated
-    _flags.recalc_wp_leash = true;
+}
+
+/// update scurve with acceleration and velocity limits
+void AC_WPNav::set_kinematic_limits(scurves &scurve_leg, const Vector3f &origin, const Vector3f &destination)
+{
+    float accel_max, vel_max;
+    const float z_length = destination.z - origin.z;
+    const float xy_length = Vector2f(destination.x - origin.x, destination.y - origin.y).length();
+    const float wp_speed_xy_cms = MAX(_wp_desired_speed_xy_cms, WPNAV_WP_SPEED_MIN);
+    const float wp_accel_xy_cmss = _pos_control.get_max_accel_xy();
+    const float wp_speed_up_cms = _pos_control.get_max_speed_up();
```

Comment:
Yes it does.

Reply:
this is because AC_WPNav use to need to slew the horizontal speed up or down when set_speed_xy() was called.  That is no longer necessary so we could actually remove the _wp_desired_speed_xy_cms variable and instead use _pos_control.get_max_speed().

Let's do this in a follow-up PR though.